### ISSUE TITLE
ICE: assertion failed: !hidden_types.has_erased_regions()

### ISSUE THREAD

**Issue by saethlin** [rust-lang/rust#116306]
I might be able to `git bisect` this tomorrow, unfortunately cargo-bisect-rustc can't help because this is a debug assertion.

```Rust
cargo +stage1 install cargo-nextest --force
```

Starts at least on this commit:
```
ca62d2c445628587660ae48013f460b08b1f5552
```

### Error output

```
thread 'rustc' panicked at compiler/rustc_trait_selection/src/traits/select/mod.rs:3119:9:
assertion failed: !hidden_types.has_erased_regions()
```

[rustc-ice-2023-10-01T04:39:24.992139512Z-887849.txt](https://github.com/rust-lang/rust/files/12776194/rustc-ice-2023-10-01T04.39.24.992139512Z-887849.txt)

**Comment by lqd** [comment]
> unfortunately cargo-bisect-rustc can't help because this is a debug assertion

I don't know about you, but I'm not sure we're getting everything we'd want out of the alt builds. I rarely hear about them, and especially not when they could have helped, like here: maybe enough time has passed by now to say debug assertions should also be turned on in the alt builds?

Anyway, I believe `cargo-nextest-0.9.59` started ICEing in release mode with 286502c9ed77e49aea5bb89cf18c5eda3a8fce92 from #107421.

**Comment by lqd** [comment]
This should have been fixed by #116452, can this issue be closed?
